src/main.ts

```diff
@@ -615,6 +617,22 @@ const launchProxyAndAttachEmitter = async (browserWindow: BrowserWindow) => {
         // don't restart the proxy if the app is shutting down or if it's already restarting
         return
       }
+
+      if (proxyRetryCount === PROXY_RETRY_LIMIT && !automaticallyFindPort) {
+        proxyRetryCount = 0
+        proxyEmitter.emit('status:change', 'offline')
```

Comment:
Setting a max retry attempt if the user opts not to find an available port automatically.